spec/design_a.md

```diff
@@ -48,3 +48,48 @@ reflect that someone wants to send tokens from Ethereum. Every subsequent
 node adds another confirmation to the peg zone state. Every BeginBlock
 invocation the peg zone checks whether any incoming Ethereum transfers have 
 reached a super-majority of confirmations and if so creates an IBC packet.
+
+## API
+
+### Ethereum Smart Contract
+
+#### update(address[] newAddress, uint64[] newPower, /*bytes32 nonce,*/ uint8[] v, bytes32[] r, bytes32[] s)
```

Comment:
So the nonce wouldn't be passed in as argument. it would be internal state of the function. It would need to be signed.

I suspect with need a `getNonce` function here. I also think nonces should be `uint32` vs bytes32